sdk/core/azure-core/src/main/java/com/azure/core/credential/BasicAuthenticationCredential.java

```diff
@@ -32,24 +25,15 @@
      * @param password basic auth password
      */
     public BasicAuthenticationCredential(String username, String password) {
-        this.username = username;
-        this.password = password;
+        String credential = username + ":" + password;
+        this.encodedCredential = Base64Util.encodeToString(credential.getBytes(StandardCharsets.UTF_8));
     }
 
     /**
      * @throws RuntimeException If the UTF-8 encoding isn't supported.
      */
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        String credential = username + ":" + password;
-        String encodedCredential;
-        try {
-            encodedCredential = Base64Util.encodeToString(credential.getBytes("UTF8"));
-        } catch (UnsupportedEncodingException e) {
-            // The encoding is hard-coded, so if it's unsupported, it needs to be fixed right here.
-            throw logger.logExceptionAsError(new RuntimeException(e));
```

Comment:
nit: we're no longer logging the exception here

Reply:
This is fine as the potential error happening before was if the `getBytes(String charset)` charset didn't exist, this can no longer happen because using `getBytes(Charset charset)` ensures the encoding exists (or at least doesn't throw a checked exception).